app/main.py

```
@@ -1,52 +1,30 @@
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
-
-
-def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+def main(book: Book, commands: list[tuple[str, str]]) -> Optional[str]:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            display = (DisplayConsole() if method_type == "console"
+                       else DisplayReverse())
+            display.display(book)
         elif cmd == "print":
-            book.print_book(method_type)
+            printer = (PrintConsole() if method_type == "console"
+                       else PrintReverse())
+            printer.print(book)
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            serializer = (SerializeJSON() if method_type == "json"
+                          else SerializeXML())
+            return serializer.serialize(book)
+        else:
+            raise ValueError(f"Unknown command {cmd}")
```

Raising ValueError for unknown commands is a sensible guard. The error message includes the unknown command which is useful for debugging; keep this behavior to avoid silent failures.